src/components/tables/TableActionsDropdown.tsx

```diff
@@ -0,0 +1,60 @@
+import { DeleteOutlined, MoreOutlined } from '@ant-design/icons';
+import { Button, Dropdown, MenuProps } from 'antd';
+import { useMemo, useState } from 'react';
+import { DeleteTableModal } from '../modals/DeleteTableModal';
+
+interface TableActionDropdownProps {
+  catalog: string;
+  schema: string;
+  table: string;
+  tableFullName: string;
+}
+
+
+export default function TableActionsDropdown({
+  catalog,
+  schema,
+  table,
+  tableFullName,
+}: TableActionDropdownProps) {
+  const [dropdownVisible, setDropdownVisible] = useState<boolean>(false);
+  const [openModal, setOpenModal] = useState<boolean>(false);
```

Comment:
You might want to extend this for other usages like `rename`

```suggestion
 enum TableActionsEnum {
    Delete
    Rename
 }
  const [action, setAction] = useState<TableActionsEnum | null>(null);
  
  // usage 
  setOpenModal(TableActionsEnum.Delete)
  <Modal open={action === TableActionsEnum.Delete} />
```